From fc7c50cae8039d2f499493b14d62de5ac6ff0f3e Mon Sep 17 00:00:00 2001
From: Auto Configured <auto.configured>
Date: Mon, 28 Sep 2015 19:07:21 +0900
Subject: [PATCH 1/1] Energy Efficient ether support


Signed-off-by: Auto Configured <auto.configured>
---
 arch/arm/mach-lm2/lm2.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-lm2/lm2.c b/arch/arm/mach-lm2/lm2.c
index a2fcdaf..7a1a331 100644
--- a/arch/arm/mach-lm2/lm2.c
+++ b/arch/arm/mach-lm2/lm2.c
@@ -42,6 +42,7 @@
 #include <mach/irqs.h>
 #include <mach/eth.h>
 #include <plat/sched_clock.h>
+#include <linux/phy.h>
 
 #include "core.h"
 
@@ -170,6 +171,7 @@ static	struct	plat_stmmacenet_data	lm2_eth_config = {
 	.has_gmac	= 1,
 	.clk_csr	= 0,
 	.enh_desc       = 1,	/* add */
+	.interface	= PHY_INTERFACE_MODE_MII,
 };
 
 static struct platform_device lm2_eth_device = {
-- 
1.7.1

