From 25225491925b220b7d3eddea4c60041f976c2fa6 Mon Sep 17 00:00:00 2001
From: Jayesh <jayesh.babu@windriver.com>
Date: Fri, 27 May 2016 13:42:58 +0900
Subject: [PATCH] 0218-Standard_Kernel_i2c_Resume


diff --git a/drivers/i2c/busses/i2c-lm2.c b/drivers/i2c/busses/i2c-lm2.c
index bb58787..657cef8 100644
--- a/drivers/i2c/busses/i2c-lm2.c
+++ b/drivers/i2c/busses/i2c-lm2.c
@@ -340,6 +340,8 @@ static int lm2_i2c_probe(struct platform_device *pdev)
 	}
 	dev_info(&pdev->dev, "I2C Driver Registered iomem = %x misc = %x IRQ = %d\n",
 		 id->iobase, id->miscbase, id->irq);
+
+        mutex_init(&id->lock);
 #if 0
 	dev_info(&pdev->dev, "EMSR=%x\n",(int)*((int *)(id->iobase+EMSR)));
 	dev_info(&pdev->dev, "ESAR=%x\n",(int)*((int *)(id->iobase+ESAR)));
diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 509c9d7..682cea0 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -312,10 +312,10 @@ static int xhci_host_suspend(struct platform_device *pdev)
 	struct xhci_hcd *xhci = hcd_to_xhci(hcd);
 	int i=0;
 	void __iomem *base;
-	base = ioremap_nocache(0x04600000, 0x500);
-	xhci_reg_save(base, &i, 0x020, 15);
-	xhci_reg_save(base, &i, 0x420,  4);
-	iounmap(base);
+	//base = ioremap_nocache(0x04600000, 0x500);
+	//xhci_reg_save(base, &i, 0x020, 15);
+	//xhci_reg_save(base, &i, 0x420,  4);
+	//iounmap(base);
 
 	base = ioremap_nocache(0x0460c000, 0x700);
 	xhci_reg_save(base, &i, 0x100, 36);
@@ -364,10 +364,10 @@ static int xhci_host_resume(struct platform_device *pdev)
                 chksum_info |= 0x100;
         }
 	i=0;
-	base = ioremap_nocache(0x04600000, 0x500);
-	xhci_reg_load(base, &i, 0x020, 15);
-	xhci_reg_load(base, &i, 0x420,  4);
-	iounmap(base);
+	//base = ioremap_nocache(0x04600000, 0x500);
+	//xhci_reg_load(base, &i, 0x020, 15);
+	//xhci_reg_load(base, &i, 0x420,  4);
+	//iounmap(base);
 
 	base = ioremap_nocache(0x0460c000, 0x700);
 	xhci_reg_load(base, &i, 0x100, 36);
-- 
1.7.1

