From 6b6f36de76f97c2a1daf44e846c2b652517a701e Mon Sep 17 00:00:00 2001
From: Auto Configured <auto.configured>
Date: Fri, 5 Dec 2014 17:25:56 +0900
Subject: [PATCH 03/18] add fxcl waikiki machine type


Signed-off-by: Auto Configured <auto.configured>
---
 arch/arm/tools/mach-types |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/arm/tools/mach-types b/arch/arm/tools/mach-types
index a10297d..6338c43 100644
--- a/arch/arm/tools/mach-types
+++ b/arch/arm/tools/mach-types
@@ -1007,3 +1007,4 @@ eco5_bx2		MACH_ECO5_BX2		ECO5_BX2		4572
 eukrea_cpuimx28sd	MACH_EUKREA_CPUIMX28SD	EUKREA_CPUIMX28SD	4573
 domotab			MACH_DOMOTAB		DOMOTAB			4574
 pfla03			MACH_PFLA03		PFLA03			4575
+lm2			MACH_LM2		LM2			5500
-- 
1.7.1

